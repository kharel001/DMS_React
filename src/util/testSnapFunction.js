import React from 'react';
import {
    configure,
    shallow
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})


export  const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

export const findByTestProps=(wrapper,val)=>{
    return wrapper.findByProps({"data-test": val})
}


  export const mockSocket = OS => {
    jest.resetModules();
    jest.doMock('../socket', () => ({
      socket:{
          emit:jest.fn(),
          on:jest.fn(),
          off:jest.fn()

      },
      emit:jest.fn(),
      on:jest.fn(),
      off:jest.fn()
    }));
  };
  
    export const mockMaterialUI = ()=>{
      jest.doMock('@material-ui/styles', () => ({
          Theme:{
            spacing:{
              unit:20
            }
          },makeStyles:jest.fn(),
          createStyles:jest.fn()
        }));
      };
    export const mockAvatar = ()=>{
      jest.resetModules();
      jest.doMock('@material-ui/core/es/Avatar/Avatar', (props) => <div >{props.children}</div>);
      };
export const mockDevice = ()=>{
    jest.resetModules();
    jest.doMock('react-device-detect', () => ({
        isIOS:false,
        isMobile:true
      }));
    };
const useHistoryFunc=jest.fn();
const listenFunc = jest.fn();
useHistoryFunc.mockReturnValue({
    listen:listenFunc
})
export const mockHistory = ()=>{
    jest.resetModules();
    // jest.doMock('react-router-dom/cjs/react-router-dom', () => ({
    //     useHistory: listenFunc
    //   }));

    };
export default function testSnapFunction(oDesc, descriptions,element){
    describe(oDesc, () => {

        afterEach(() => {
            jest.clearAllMocks();
          });

        test(descriptions, () => {
            const tree = shallow(element);
            expect(tree).toMatchSnapshot();
        });
    
    })
}

