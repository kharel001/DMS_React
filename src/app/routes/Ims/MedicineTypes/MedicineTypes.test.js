import React from 'react';
import MedicineTypes from './MedicineTypes';
import {
    configure,
    shallow,
    mount
} from 'enzyme';
import types from '../../../../JSONFiles/medicineTypes';

import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
import { mockMaterialUI } from '../../../../util/testSnapFunction';
configure({
    adapter: new EnzymeAdapter
})


const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

describe('<MedicineTypes />', () => {
  let wrapper;

  let store;

  beforeEach(() => {
    
  mockMaterialUI();
  })

  afterEach(() => {
    jest.clearAllMocks();
  });
 

it("MedicineTypes snapshot", () => {
        // let wrapper = setup(dummy_data);
    wrapper = shallow( < MedicineTypes  />)

        expect(wrapper).toMatchSnapshot();

  
})
it("MedicineTypes comp", () => {
    // let wrapper = setup(dummy_data);
wrapper = shallow( < MedicineTypes  />)

let cardComp=findByTestAttr(wrapper, "cardComp")
expect(cardComp).toHaveLength(types.length)

// expect(rowsComp).toHaveLength(rows.length)

})
})