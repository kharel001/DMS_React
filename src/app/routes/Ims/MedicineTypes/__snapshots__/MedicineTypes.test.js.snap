// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MedicineTypes /> MedicineTypes snapshot 1`] = `ShallowWrapper {}`;
